.github/workflows/main.yml

```diff
@@ -25,40 +25,55 @@ jobs:
     steps:
     - uses: actions/checkout@v2
 
+    - name: Set up JDK 1.11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 1.11
+
     - name: Download Ktlint
       run: |
         KTLINT="0.37.0"
         echo Using Ktlint $KTLINT
         curl -sSLO https://github.com/pinterest/ktlint/releases/download/$KTLINT/ktlint
         chmod a+x ktlint
-    
-    -  name: Download buf
-       run: |
+
+    - name: Download buf
+      run: |
         BUFVERSION="v0.15.0"
         echo Using Bufbuild version $BUFVERSION
         curl -sSL https://github.com/bufbuild/buf/releases/download/$BUFVERSION/buf-$(uname -s)-$(uname -m) > buf
         chmod a+x buf
-        
+
+    - name: Download Checkstyle
+      run: |
+        CHECKSTYLE="8.37"
+        echo Using Checkstyle version $CHECKSTYLE
+        curl -sSLO https://github.com/checkstyle/checkstyle/releases/download/checkstyle-$CHECKSTYLE/checkstyle-$CHECKSTYLE-all.jar
+        chmod a+x checkstyle-$CHECKSTYLE-all.jar
+
     - name: Kotlin lint check
       run: ./ktlint --android domain/src/**/*.kt utility/src/**/*.kt data/src/**/*.kt app/src/**/*.kt testing/src/**/*.kt && echo "Lint completed successfully"
-    
-    -  name: Protobuf lint check
-       run: ./buf check lint --input=model/src/main/proto --input-config buf.yaml && echo "Protobuf lint check completed successfully"
-     
+
+    - name: Protobuf lint check
+      run: ./buf check lint --input=model/src/main/proto --input-config buf.yaml && echo "Protobuf lint check completed successfully"
+
+    - name: Java lint check
+      run: java -jar checkstyle-8.37-all.jar -c=checkstyle.xml app/src/ data/src/ domain/src/ utility/src/ testing/src/ && echo "Lint completed successfully"
+
```

Comment:
Restructure these linters to run in the order mentioned above - Java, Kotlin, Protobuf

Reply:
done

Reply:
resolving on behalf of Vinita